### ISSUE TITLE
[Bug] Label mapping configs do not work when set from Helm values.yaml

### ISSUE THREAD

**Issue by Jeinhaus** [kubecost/kubecost#3722]
### Kubecost Helm Chart Version

v2.4.1

### Kubernetes Version

v1.30.4

### Kubernetes Platform

EKS

### Description

Defining and enabling `kubecostProductConfigs.labelMappingConfigs` does not work.
It seems, that the generated configmap (`app-configs`) contain the correct information, but those are not reflected in the kubecost UI. The UI still shows the default values for all the label mappings.
We tried both our label format as well as the same label with all special characters replaced by `_` as reflected in the Prometheus metrics.

Trying the missing setting from https://github.com/kubecost/cost-analyzer-helm-chart/issues/3264 did not work either.

### Steps to reproduce

1. Set the `kubecostProductConfigs.labelMappingConfigs`  to

  ```yaml
   labelMappingConfigs:
     enabled: true
     team_label: platform/tenant
     environment_label: platform/environment-name
  ```
2. Deploy the chart
3. Open the settings page and check the "Labels" section

### Expected behavior

I would expect the labels defined in `kubecostProductConfigs.labelMappingConfigs` to be set in the UI.

### Impact

Not too big, since we can set these in the UI manually.

### Troubleshooting

- [X] I have read and followed the [issue guidelines](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/ISSUE_GUIDELINES.md) and this is a bug impacting only the Helm chart.
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by jessegoodier** [comment]
Thanks for the detail. we will take a look.

**Comment by jessegoodier** [comment]
I didn't notice this earlier, but you have a / in your label.
These are converted to _ in the metrics.
If you go to the UI and do a filter for label = platform it should autocomplete the available labels and that will be what you can use in the helm config.

Does this make sense?
@KarolTegha FYI

**Comment by Jeinhaus** [comment]
@jessegoodier thanks for looking into this.

As mentionend above, we already tried this:

>  We tried both our label format as well as the same label with all special characters replaced by _ as reflected in the Prometheus metrics.

This did not work either sadly :-/

Also, setting these labels with the `/` in the UI did actually work. So it's a bit confusing what these labels should actually look like.

**Comment by jessegoodier** [comment]
Edit- I believe I found the issue. Will send an update when I have more detail.

**Comment by jessegoodier** [comment]
Plase see edit above, I think we have a repro.

**Comment by Jeinhaus** [comment]
> Plase see edit above, I think we have a repro.

Any update on this? @jessegoodier

**Comment by jessegoodier** [comment]
Yes, this is certainly a bug when using aggregator as a statefulset. I can give you a workaround now if it is urgent. 
Otherwise we will get it fixed in the next patch or 2.5.

**Comment by Jeinhaus** [comment]
> Yes, this is certainly a bug when using aggregator as a statefulset. I can give you a workaround now if it is urgent. Otherwise we will get it fixed in the next patch or 2.5.

I would appreciate a workaround, even if it's not very urgent. 
Is the bug related to https://github.com/kubecost/cost-analyzer-helm-chart/issues/3264#issuecomment-2464154808 as well? Would be great to have both working again.

**Comment by jessegoodier** [comment]
I'm not sure I understand the #3264 relation.
for this, there is a configmap that is not being mounted to the aggregator statefulset.
We will get a patch out next week. 

if you look at:
`sts/kubecost-aggregator -- cat /var/configs/apiconfig.json`

and edit that to your needs, you can then upload it with a hack like:
`cat "$file" | kubectl exec -i -n "$NAMESPACE" $kubecost_aggregator -c aggregator -- sh -c "cat > /var/configs/apiconfig.json")`

**Comment by Jeinhaus** [comment]
Thanks for the workaround and great that a patch is on the way already.
The #3264 relation was just added, because multiple of those settings in the values.yaml section seem to not work (correctly), so I thought this might be related. But that was just guesswork without any indication. Sorry.
